icu4c/source/tools/genrb/genrb.cpp

```diff
@@ -163,7 +163,7 @@ main(int argc,
     }
     if(options[FORMAT_VERSION].doesOccur) {
         const char *s = options[FORMAT_VERSION].value;
-        if(uprv_strlen(s) != 1 || (s[0] < '1' && '3' < s[0])) {
+        if(uprv_strlen(s) != 1 || (s[0] < '1' || s[0] > '3')) {
```

Comment:
let's preserve the intention of the "is the value outside the range" check
```suggestion
        if(uprv_strlen(s) != 1 || (s[0] < '1' || '3' < s[0])) {
```

Reply:
Done.